backend/endpoints/auth.js

```diff
@@ -95,72 +94,56 @@ app.post("/check-credentials", async (req, res) => {
   });
 
   /***
-   * Attempts to log in using the provided credentials.
-   * Returns 200 for successful log in attempt and a 401 otherwise.
+   * Gets the user's GitHub authentication token and returns it.
    */
-  app.post("/login", async (req, res) => {
-    const { userHandle, password } = req.body;
-    const user = await prisma.user.findUnique({
-      where: { userHandle: userHandle },
-    });
-
-    if (!user) {
-      res.status(401).json();
-      return;
-    }
+  app.post("/token-auth", async (req, res) => {
+    const GITHUB_ACCESS_TOKEN_URL =
+      "https://github.com/login/oauth/access_token";
 
+    const { clientID, clientSecret, code } = req.body;
 
-/***
- * Gets the user's GitHub authentication token and returns it.
- */
-app.post("/token-auth", async (req, res) => {
-  const GITHUB_ACCESS_TOKEN_URL = "https://github.com/login/oauth/access_token";
+    const accessTokenParams = new URLSearchParams({
+      client_id: clientID,
+      client_secret: clientSecret,
+      code: code,
+    });
 
-  const { clientID, clientSecret, code } = req.body;
+    const fetchURL = `${GITHUB_ACCESS_TOKEN_URL}?${accessTokenParams}`;
 
-  const accessTokenParams = new URLSearchParams({
-    client_id: clientID,
-    client_secret: clientSecret,
-    code: code,
-  });
+    const response = await fetch(fetchURL, {
+      headers: {
+        method: "POST",
+        Accept: "application/json",
+      },
+    });
 
-  const fetchURL = `${GITHUB_ACCESS_TOKEN_URL}?${accessTokenParams}`;
+    const token = await response.json();
 
-  const response = await fetch(fetchURL, {
-    headers: {
-      method: "POST",
-      Accept: "application/json",
-    },
+    if (token.access_token) {
+      // If successful, hash and store token.
+      res.status(200).json(token);
+    } else {
+      // Invalid or already fuffilled request for access token.
+      if (true) {
+        // Already fuffilled.
+        // Fetch access code from database and return.
+        res.status(200).json();
+      } else {
+        // Invalid request.
+      }
```

Comment:
Do you plan on iterating on this?

Reply:
Returned code for error handling in frontend.